osdconfig/proto/config.proto

```diff
@@ -0,0 +1,120 @@
+syntax = "proto3";
+
+// this package defines the configuration parameters and interfaces
+// requires to run portwork services
+package proto;
+
+service Spec {
+    rpc GetGlobalSpec(Empty) returns (GlobalConfig) {}
+    rpc SetGlobalSpec(GlobalConfig) returns (Ack) {}
+    rpc GetClusterSpec(Empty) returns (ClusterConfig) {};
+    rpc SetClusterSpec(ClusterConfig) returns (Ack) {};
+    rpc GetNodeSpec(NodeID) returns (NodeConfig) {};
+    rpc SetNodeSpec(NodeConfig) returns (Ack) {};
+}
+
+message Empty {}
+message Ack {
+    int64 n = 1;
+}
+message NodeID {
+    string i_d = 1;
+}
+
+// Global Conf contains all the configuration data
+message GlobalConfig {
+    ClusterConfig cluster_conf = 1;
+    NodesConfig nodes_conf = 2;
+}
+
+message NodesConfig {
+    map<string, NodeConfig> node_conf = 1;
+}
+
+message NodeConfig {
+    string node_id = 1;
+    NetworkConfig network = 2;
+    SecretsConfig secrets = 3;
+    StorageConfig storage = 4;
+}
+
+message KVDBConfig {
+    string username = 1;
```

Comment:
Can you also add comments for each field so that it would show up in the swagger docs? I can help you with that.